training/src/app/modules/chuck-norris/components/joke-list/joke-list.component.ts

```diff
@@ -9,14 +12,19 @@ import { Component, Input, OnInit } from '@angular/core';
 })
 export class JokeListComponent implements OnInit {
 
+	constructor(private store: Store<AppState>) {
+	}
+
 	@Input() public jokes: Joke[];
 	public jokeClickedId: string;
 	public isTableExpanded: boolean            = false;
 	public jokeList: MatTableDataSource<Joke>  = new MatTableDataSource();
 	public displayedJokesColumnsList: string[] = ['id', 'category', 'date_created'];
 
 	ngOnInit(): void {
-		this.jokeList.data = this.jokes;
+		this.store.select(getSearchListJokesSelector).subscribe(
```

Comment:
what about unsubscribe?

Reply:
also, this component is "simple and stupid", just a table.
its better to do it the chuck main stuff and pass the values (like before).
this way this component doesnt need to know about store